xbmc/guilib/guiinfo/VideoGUIInfo.cpp

```diff
@@ -440,7 +440,14 @@ bool CVideoGUIInfo::GetLabel(std::string& value, const CFileItem *item, int cont
         value = tag->m_streamDetails.GetAudioLanguage();
         return true;
       case LISTITEM_SUBTITLE_LANGUAGE:
-        value = tag->m_streamDetails.GetSubtitleLanguage();
+        if (g_application.GetAppPlayer().GetSubtitleVisible())
```

Comment:
I don't think this is the correct approach, you're hiding the error which is the fact the subtitle stream info is still populated (and with the wrong language) if the player doesn't have any active subtitle. I'd do it like below:

```
diff --git a/xbmc/cores/VideoPlayer/VideoPlayer.cpp b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
index d4b52110c3..3abc882e1c 100644
--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
@@ -3245,6 +3245,8 @@ void CVideoPlayer::SetSubtitleVisibleInternal(bool bVisible)

   if (m_pInputStream && m_pInputStream->IsStreamType(DVDSTREAM_TYPE_DVD))
     std::static_pointer_cast<CDVDInputStreamNavigator>(m_pInputStream)->EnableSubtitleStream(bVisible);
+
+  CServiceBroker::GetDataCacheCore().SignalSubtitleInfoChange();
 }

 std::shared_ptr<TextCacheStruct_t> CVideoPlayer::GetTeletextCache()
@@ -5036,12 +5038,14 @@ void CVideoPlayer::GetSubtitleStreamInfo(int index, SubtitleStreamInfo &info)
 {
   CSingleLock lock(m_content.m_section);

-  if (index == CURRENT_STREAM)
+  if (index == CURRENT_STREAM && GetSubtitleVisible())
     index = m_content.m_subtitleIndex;

   if (index < 0 || index > GetSubtitleCount() - 1)
   {
     info.valid = false;
+    info.language = "";
+    info.flags = StreamFlags::FLAG_NONE;
     return;
   }
```

If you want to show "None" or "Disabled" or another alternative string instead of an empty string that should be handled by the skin when setting the label using some sort of conditional visibility or skin variables (`String.IsEmpty` being the starting point). Not sure how to do this on the skin but maybe @ronie can help.

Reply:
Updated accordingly.

Skin change likely won't work.

Reply:
Yeah, skin changes don't work. But testing it, seems like it's not a show stopper to just show it as "". Might be clearer with another label, but probably not worth the effort.